staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go

```diff
@@ -426,11 +426,36 @@ func (g *Cloud) AddAliasToInstance(nodeName types.NodeName, alias *net.IPNet) er
 // Gets the named instances, returning cloudprovider.InstanceNotFound if any
 // instance is not found
 func (g *Cloud) getInstancesByNames(names []string) ([]*gceInstance, error) {
+	foundInstances, err := g.getFoundInstanceByNames(names)
+	if err != nil {
+		return nil, err
+	}
+	var failed []string
+	foundInstanceNames := sets.NewString()
+	for _, instance := range foundInstances {
```

Comment:
My thought is that since getFoundInstanceByNames returns all the found instances, just compare the len(names) and len(foundInstances), then you can be sure whether all instances are found or not, and return either the list or error.

Reply:
This is done so the instances that caused error can be logged.

Reply:
you can log in getFoundInstanceByNames instead of here to save the loops here.

Reply:
This is there as a remnant from when it still returned instance or error. I've moved it.